packages/redeem_solver/src/ika_client.ts

```diff
@@ -0,0 +1,176 @@
+import {
+	IkaClient as SdkIkaClient,
+	IkaTransaction,
+	getNetworkConfig,
+	Curve,
+	SignatureAlgorithm,
+	Hash,
+	SessionsManagerModule,
+	CoordinatorInnerModule,
+	createUserSignMessageWithCentralizedOutput,
+	type IkaConfig,
+} from "@ika.xyz/sdk";
+import type { SuiNet } from "@gonative-cc/lib/nsui";
+import type { SuiClient as MystenClient } from "@mysten/sui/client";
+import {
+	Transaction,
+	type TransactionArgument,
+	type TransactionObjectArgument,
+} from "@mysten/sui/transactions";
+
+export interface IkaClient {
+	initialize(): Promise<void>;
+	getIkaCoin(owner: string): Promise<string>;
+	getLatestNetworkEncryptionKeyId(): Promise<string>;
+	getCoordinatorId(): string;
+	getPresignCapId(presignId: string): Promise<string>;
+
+	createUserSigMessage(
+		dwalletId: string,
+		presignId: string,
+		message: Uint8Array,
+	): Promise<Uint8Array>;
+
+	requestGlobalPresign(
+		tx: Transaction,
+		ikaCoin: TransactionObjectArgument,
+		suiCoin: TransactionObjectArgument,
+		dwalletNetworkEncryptionKeyId: string,
+	): TransactionObjectArgument;
+
+	createSessionIdentifier(tx: Transaction): TransactionArgument;
+
+	decodePresignRequestEvent(bcs: string): { presign_id: string };
+	decodeSignRequestEvent(bcs: string): { sign_id: string };
+}
+
+export class IkaClientImp implements IkaClient {
+	private ikaClient: SdkIkaClient;
+	private ikaConfig: IkaConfig;
+
+	constructor(
+		network: SuiNet,
+		private mystenClient: MystenClient,
+	) {
+		// this is needed because ika do not support devnet as of now
+		const ikaNetwork = network === "mainnet" ? "mainnet" : "testnet";
+		this.ikaConfig = getNetworkConfig(ikaNetwork);
+
+		this.ikaClient = new SdkIkaClient({
+			suiClient: this.mystenClient,
+			config: this.ikaConfig,
+		});
+	}
+
+	async initialize(): Promise<void> {
+		if (this.ikaClient.initialize) {
+			await this.ikaClient.initialize();
+		}
+	}
+
+	getCoordinatorId(): string {
+		return this.ikaConfig.objects.ikaDWalletCoordinator.objectID;
+	}
+
+	async getIkaCoin(owner: string): Promise<string> {
+		const coins = await this.mystenClient.getCoins({
+			owner: owner,
+			coinType: `${this.ikaConfig.packages.ikaPackage}::ika::IKA`,
+			limit: 5,
+		});
+		// TODO: for now lets just take the first one
```

Comment:
what if the value is too small?
let's add a task to rework it (add it to the epic)